packages/backend/src/open_payments/payment_pointer/middleware.ts

```diff
@@ -5,16 +5,18 @@ export function createPaymentPointerMiddleware() {
     ctx: AppContext,
     next: () => Promise<unknown>
   ): Promise<void> => {
-    const paymentPointerService = await ctx.container.use(
-      'paymentPointerService'
-    )
-    const paymentPointer = await paymentPointerService.getByUrl(
-      `https://${ctx.request.host}/${ctx.params.paymentPointerPath}`
-    )
-    if (!paymentPointer) {
-      ctx.throw(404)
+    const url = `https://${ctx.request.host}/${ctx.params.paymentPointerPath}`
+    const config = await ctx.container.use('config')
+    if (url !== config.paymentPointerUrl) {
```

Comment:
In which situations will this equality happen?

Reply:
When this is an Open Payments resource request for the RS's own payment pointer, which will now/soon be happening when a separate AS is querying RS's client payment pointer `/jwks.json`